app/models/cherrypick_task/control_locator.rb

```diff
@@ -40,13 +45,18 @@ class CherrypickTask::ControlLocator
   # @param total_wells [Integer] The total number of wells on the plate
   # @param num_control_wells [Integer] The number of control wells to lay out
   # @param wells_to_leave_free [Enumerable] Array or range indicating the wells to leave free from controls
-  def initialize(batch_id:, total_wells:, num_control_wells:, wells_to_leave_free: [])
+
+  # rubocop:disable Metrics/ParameterLists
+  def initialize(batch_id:, total_wells:, num_control_wells:, template:, control_source_plate:, wells_to_leave_free: [])
```

Comment:
Again, if we have so many parameters, maybe we should turn them into a structured object.  Either a hash with expected keys for a quick workaround, or a wrapper object with named attributes for a better solution.